tests/components/bond/test_config_flow.py

```diff
@@ -326,13 +327,12 @@ async def test_zeroconf_already_configured(hass: core.HomeAssistant):
                 type="mock_type",
             ),
         )
+        await hass.async_block_till_done()
 
     assert result["type"] == "abort"
     assert result["reason"] == "already_configured"
     assert entry.data["host"] == "updated-host"
-
-    await hass.async_block_till_done()
-    assert len(mock_setup_entry.mock_calls) == 0
+    assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
when abort reason `already_configured` is reached in config flow the config entry is reloaded, which calls `async_setup_entry`. Prior this change `await hass.async_block_till_done()` was called outside the `with` block, those did not call the mocked async_setup_entry